### ISSUE TITLE
Barri dungeon generated incorrectly

### ISSUE THREAD

**Issue by Pegwald** [aura-project/aura#271]
I have a theory that Ore patches can replace those four switch props in a room that normally opens the boss door. That must be the case in this run or I did simply not get any boss key at all. I got to the room with the boss door, but instead of cones, I find those ore patches. No switches, no key, just 3 piles of ore.

I remember I might have reported this issue earlier, but here's what the channel log has to offer this time:
2016-05-12 02:35 [Debug] - Dungeon: Creating 'bangor_barri_dungeon', item id: 2000, seed: 2090655185, floorPlan: 5
2016-05-12 02:35 [Warning] - Dungeon.InitFloorRegion: No locked place in last section of 'bangor_barri_dungeon'.

**Comment by exectails** [comment]
You're right, that version of Barri doesn't get a boss key. The boss door can still be opened, because the server detected that problem, but there's no key. @Tachiorz We "finally" have an example.

**Comment by exectails** [comment]
![ss 2016-05-15 at 09 34 23](https://cloud.githubusercontent.com/assets/5593920/15272651/3ee8826a-1a80-11e6-8065-6407bc23d4c0.png)

**Comment by exectails** [comment]
Another one, @Tachiorz.
```text
[Debug] - Dungeon: Creating 'g1_30_bangor_barri_dungeon', item id: 73012, seed: 32201302, floorPlan: 1715256805
[Warning] - Dungeon.InitFloorRegion: No locked place in last section of 'g1_30_bangor_barri_dungeon'.
```

**Comment by exectails** [comment]
@Tachiorz Any updates?

**Comment by Tachiorz** [comment]
@exectails Thanks for remainder, no updates. But I really have to deal with it soon, this bug is ancient.

**Comment by Sairii** [comment]
http://i.imgur.com/3TEfEiA.jpg

happens quite often in alby dungeon.
